src/app/OperationalDeviceProxy.cpp

```diff
@@ -270,14 +270,19 @@ void OperationalDeviceProxy::DequeueConnectionCallbacks(CHIP_ERROR error)
         }
     }
 
+    auto * exchangeMgr = GetExchangeManager();
+    auto sessionHandle = mSecureSession.Get();
     while (successReady.mNext != &successReady)
     {
         Callback::Callback<OnDeviceConnected> * cb = Callback::Callback<OnDeviceConnected>::FromCancelable(successReady.mNext);
 
         cb->Cancel();
         if (error == CHIP_NO_ERROR)
         {
-            cb->mCall(cb->mContext, this);
+            // We know that we for sure have the SessionHandler in the successful case.
+            VerifyOrDie(exchangeMgr);
+            VerifyOrDie(sessionHandle.HasValue());
+            cb->mCall(cb->mContext, exchangeMgr, sessionHandle.Value());
         }
```

Comment:
Where is the logic to actually call into `CASESessionManager` (through some kind of delegate) to actually free up this object?

Reply:
This part still needs to happen as a part of this PR, which I will do assuming there are no objections to the `OnDeviceConnected` API changes and the interim solution of `DeviceProxySession` from folks outside of Google